### PR TITLE
Added ability to delete filters and segments, validations for filters

### LINKED ISSUE #16: Misc Issues
- [x] Add functionality to deletion of individual filter blocks.
- [x] Add ability to delete segments.
- [x] Add validations to datatypes. Eg. `users_count` cannot be a string.

### PR DESCRIPTION
**Pull request by AbhayVAshokan**
Fixes #16
- Added the functionality to delete segments.
- Added the functionality to delete individual filter blocks.
- Added datatype validations for dropdown items.
- Updated neetoUI to v3.3.6.
- Updated neetoIcons to v1.8.30.
- Fixed minor bugs in FilterDropdown component.

Demo: https://vimeo.com/698550820/56022b2939

### CODE REVIEW

**Comment by AbhayVAshokan** [comment]
@Amaljith-K _a please review.

File: src/components/Delete.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { useSegment, ACTION_TYPES } from "contexts/segment";
+import { Delete as DeleteIcon } from "neetoicons";
+import { Alert, Button } from "neetoui";
+import { isNil } from "ramda";
+
+const Delete = ({ onDelete }) => {
+  const [isDeleteAlertOpen, setIsDeleteAlertOpen] = useState(false);
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [segment, dispatchSegment] = useSegment();
+  const isUpdate = !isNil(segment.id);
```
Comment (Amaljith-K) [inline src/components/Delete.jsx:12]:
```suggestion
  const isUpdate = !segment.id;
```

File: src/components/FilterBlock.jsx [hunk 2]
```diff
@@ -2,23 +2,32 @@ import { NODES_WITHOUT_INPUT } from "constants";
...
 import React from "react";
 
+import { useSegment, ACTION_TYPES } from "contexts/segment";
 import { Close } from "neetoicons";
 import { Tooltip, Typography } from "neetoui";
 import { formatDate, formatDateAndTime, humanize } from "utils";
 
-const FilterBlock = ({ condition }) => {
+const FilterBlock = ({ condition, groupIndex, conditionsIndex }) => {
+  const [, dispatchSegment] = useSegment();
```
Comment (Amaljith-K) [inline src/components/FilterBlock.jsx:11]:
```suggestion
  const dispatchSegment = useSegment()[1];
```

File: src/components/FilterDropdown/index.jsx [hunk 3]
```diff
@@ -30,17 +30,17 @@ const FilterDropdown = ({
...
-  };
-
   const handleClick = () => {
-    isFilterDropdownOpen ? onClose() : setIsFilterDropdownOpen(true);
+    if (isFilterDropdownOpen) {
+      setSelectedNode(node);
+      setIsFilterDropdownOpen(false);
+    } else {
+      setIsFilterDropdownOpen(true);
+    }
```
Comment (Amaljith-K) [inline src/components/FilterDropdown/index.jsx:43]: This is safer from race conditions
```suggestion
    setIsFilterDropdownOpen(isOpen => {
      if (isOpen) setSelectedNode(node);
      return !isOpen;
    });
```

File: src/components/FilterDropdown/utils.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  if (isEmpty(value)) {
+    return `${label} is a required field.`;
+  }
+
+  if (type in isValid && !isValid[type]) {
+    return `${label} should be a ${type}.`;
+  }
+
+  return "";
+};
```
Comment (Amaljith-K) [inline src/components/FilterDropdown/utils.js:20]: https://vimeo.com/698874116/f1a11de972

**Review by Amaljith-K — CHANGES_REQUESTED** [review]
@AbhayVAshokan _a

**Comment by AbhayVAshokan** [comment]
@Amaljith-K _a please have a look: https://vimeo.com/698882285/abe7d70cc7

File: example/src/App.jsx [hunk 5]
```diff
@@ -35,6 +35,10 @@ const App = () => {
     await new Promise(resolve => setTimeout(resolve, 1000));
   };
 
+  const handleDelete = async () => {
+    await new Promise(resolve => setTimeout(resolve, 1000));
+  };
```
Comment (Amaljith-K) [inline example/src/App.jsx:40]:
```suggestion
  const handleDelete = () => new Promise(resolve => setTimeout(resolve, 1000));
```
There is no need for additional async await wrap when you are already returning a promise.

**Review by Amaljith-K — APPROVED** [review]
LGTM except for one minor comment.
Please make the changes and merge

**Pull request merged**